graph/src/data/subgraph/mod.rs

```diff
@@ -451,37 +451,53 @@ pub struct Graft {
     pub block: BlockNumber,
 }
 
+macro_rules! gbi {
+    ( $( $x:expr ),* ) => {
+        {
+            vec![
+                $(
+                    SubgraphManifestValidationError::GraftBaseInvalid($x),
+                )*
+            ]
+        }
+    };
+}
```

Comment:
I wouldn't use a macro for this.

Reply:
Why not? Do you want to avoid people just using this enum variant by default?

Do you think [this](https://github.com/graphprotocol/graph-node/pull/3627/commits/089302835981ae1db4b8999123f4d9c5a362192c) looks better?

Reply:
You can have just a function that takes a vec of `String`. A macro is kind of a big hammer.

Reply:
But I can see that you want the single parameter case to not need a wrapping vec, in that aspect the macro is nice.

Reply:
This addresses the issue, but I think it can become cleaner https://github.com/graphprotocol/graph-node/pull/3627/commits/87839c489a1c1ed88bd2ea87138d72eaf8662849

Maybe it's fine for now.

Reply:
I've cleaned it more according to what we've spoke on Slack 🙂